Test/Get-WhiskeyTasks.Tests.ps1

```diff
@@ -63,15 +51,35 @@ Describe 'Get-WhiskeyTasks' {
         {
             if( $expectedTasks.ContainsKey($task.Name) )
             {
-                $task.CommandName | should -Be $expectedTasks[$task.Name]
+                $task.CommandName | Should -Be $expectedTasks[$task.Name].CommandName
+            }
+        }
+
+        foreach( $taskName in $expectedTasks.Keys )
+        {
+            $expectedTask = $expectedTasks[$taskName]
+            $actualTask = $tasks | Where-Object { $_.Name -eq $taskName } 
+
+            if( $expectedTask.Obsolete )
+            {
+                $actualTask | Should -BeNullOrEmpty -Because ('should return "{0}" task' -f $taskName)
+            }
+            else
+            {
+                $actualTask | Should -Not -BeNullOrEmpty -Because ('should not return obsolete "{0}" task' -f $taskName)

```

Comment:
Are these two reversed? When `Should -BeNullOrEmpty` then I would think it's because it "_should **not** return ... task_".

Reply:
The reasons passed to `-Because` were switched.